### PR TITLE
Feat: added helm install custom function

### PR DESCRIPTION
**Pull request by vikramdevtron**
# Description

Allow users to install custom charts, added new service which received ref chart in byte array, and extract and install in helm.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Helm install custom chart received from byte array.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [x] I have tested it for all user roles
* [ ] I have added all the required unit/api test cases

### CODE REVIEW

File: pkg/service/HelmAppServiceWrapper.go [hunk 1]
```diff
@@ -30,6 +30,17 @@ func NewApplicationServiceServerImpl(logger *zap.SugaredLogger, chartRepositoryL
 	}
 }
 
+func (impl *ApplicationServiceServerImpl) HelmInstallCustom(ctx context.Context, req *client.HelmInstallCustomRequest) (*client.HelmInstallCustomResponse, error) {
```
Comment (manish-agrawal-ai) [inline pkg/service/HelmAppServiceWrapper.go:33]: function can be renamed from HelmInstallCustom to InstallReleaseWithCustomChart

File: pkg/service/HelmAppService.go [hunk 2]
```diff
@@ -1016,11 +1038,63 @@ func (impl HelmAppServiceImpl) getHelmClient(clusterConfig *client.ClusterConfig
...
+	if err != nil {
+		impl.logger.Errorw("error on helm install custom", "err", err)
+		return false, err
+	}
+	err = writer.Close()
+	if err != nil {
+		impl.logger.Errorw("error on helm install custom", "err", err)
+		return false, err
+	}
+	err = os.MkdirAll(chartWorkingDirectory, os.ModePerm)
```
Comment (manish-agrawal-ai) [inline pkg/service/HelmAppService.go:1068]: os.MkdirAll(chartWorkingDirectory, os.ModePerm) should be call one time only (may be in NewHelmAppServiceImpl)

File: pkg/service/HelmAppService.go [hunk 3]
```diff
@@ -1016,11 +1038,63 @@ func (impl HelmAppServiceImpl) getHelmClient(clusterConfig *client.ClusterConfig
...
+	helmClientObj, err := impl.getHelmClient(releaseIdentifier.ClusterConfig, releaseIdentifier.ReleaseNamespace)
+	if err != nil {
+		impl.logger.Errorw("error on helm install custom", "err", err)
+		return false, err
+	}
+	var b bytes.Buffer
+	writer := gzip.NewWriter(&b)
+	_, err = writer.Write(request.Chunk.Content)
+	if err != nil {
+		impl.logger.Errorw("error on helm install custom", "err", err)
```
Comment (manish-agrawal-ai) [inline pkg/service/HelmAppService.go:1060]: error on helm install custom -> change to error on helm install custom while writing chartContent
(same comment for all places)

File: pkg/service/HelmAppService.go [hunk 4]
```diff
@@ -1016,11 +1038,63 @@ func (impl HelmAppServiceImpl) getHelmClient(clusterConfig *client.ClusterConfig
...
+		ReleaseName: releaseIdentifier.ReleaseName,
+		Namespace:   releaseIdentifier.ReleaseNamespace,
+		ValuesYaml:  request.ValuesYaml,
+		ChartName:   referenceChartDir,
+	}
+
+	//	impl.logger.Debug("Upgrading release with chart info")
+	_, err = helmClientObj.InstallChart(context.Background(), chartSpec)
+	if err != nil {
+		impl.logger.Errorw("Error in upgrade release with chart info", "err", err)
```
Comment (manish-agrawal-ai) [inline pkg/service/HelmAppService.go:1094]: log is wrong

File: grpc/applist.proto [hunk 5]
```diff
@@ -249,4 +250,18 @@ message RollbackReleaseRequest {
 
 message TemplateChartResponse {
   string generatedManifest = 1;
+}
+
+message HelmInstallCustomRequest {
+    string valuesYaml = 1;
+    Chunk chunk = 2;
```
Comment (manish-agrawal-ai) [inline grpc/applist.proto:257]: rename chunk to chartContent

**Review by manish-agrawal-ai — COMMENTED** [review]
please check comments

File: grpc/applist.proto (hunk 5 above)
Comment (vikramdevtron) [reply grpc/applist.proto]: done

File: pkg/service/HelmAppService.go (hunk 4 above)
Comment (vikramdevtron) [reply pkg/service/HelmAppService.go]: done

File: pkg/service/HelmAppService.go (hunk 3 above)
Comment (vikramdevtron) [reply pkg/service/HelmAppService.go]: done

File: pkg/service/HelmAppService.go (hunk 2 above)
Comment (vikramdevtron) [reply pkg/service/HelmAppService.go]: done

File: pkg/service/HelmAppService.go [hunk 6]
```diff
@@ -49,19 +57,33 @@ type HelmAppService interface {
...
+		randSource: rand.NewSource(time.Now().UnixNano()),
 	}
 }
+func (impl HelmAppServiceImpl) CleanDir(dir string) {
+	err := os.RemoveAll(dir)
+	if err != nil {
+		impl.logger.Warnw("error in deleting dir ", "dir", dir)
+	}
+}
+func (impl HelmAppServiceImpl) GetDir() string {
```
[the remark this answers is not in the document]
Comment (vikramdevtron) [reply pkg/service/HelmAppService.go]: done

File: pkg/service/HelmAppServiceWrapper.go (hunk 1 above)
Comment (vikramdevtron) [reply pkg/service/HelmAppServiceWrapper.go]: done

**Review by manish-agrawal-ai — APPROVED** [review]
approved

**Pull request merged**